test/test_handshake.cpp

```diff
@@ -294,8 +295,9 @@ void run_client_surprise_server(unit_test::TestContext& test_context, const std:
         server.run();
     });
 
+    auto socket_provider = std::make_shared<websocket::DefaultSocketProvider>(client_logger, "");
     Client::Config client_config;
-    client_config.logger = &client_logger;
```

Comment:
How did this work without crashing in the past? 
Oh, it looks like we don't use this file in our tests... (it's not in any of the CMakeLists.txt files)